backend/src/DatabaseHandler.py

```diff
@@ -0,0 +1,122 @@
+        """
+        Thread to handle the pocketbase database communication.
+        The Thread is subscribed to the CommandMessage
+        collection to wait for commands created in the front end. 
+        The handler can also send telemetry data to the database
+        to be read by the front end.
+        """
+        logger.info("DatabaseHandler initializing")
+        DatabaseHandler.thread_workq = thread_workq
+        DatabaseHandler.send_message_workq = message_handler_workq
+        DatabaseHandler.thread_name = thread_name
+        DatabaseHandler.client = Client("http://10.13.141.121:8090/")
+        DatabaseHandler.client.collection('CommandMessage').subscribe(DatabaseHandler._handle_command_callback)
+        logger.success(f"Successfully started {thread_name} thread")
+
+    @staticmethod
+    def _handle_command_callback(document: MessageData):
+        """
+        Whenever a new entry is created in the CommandMessage 
+        collection, this function is called to handle the
+        command and forward it to the serial port.
+
+        Args:
+            document (MessageData): the change notification from the database.
+        """
+
+        logger.info("Received new command from the database")
+        logger.debug(f"Record command: {document.record.command}")
+        DatabaseHandler.send_message_workq.put(
+            WorkQ_Message(
+                DatabaseHandler.thread_name,
+                'all_serial', 
+                THREAD_MESSAGE_SERIAL_WRITE, 
+                (document.record.command,
+                 document.record.target,
+                 document.record.command_param,
+                 document.record.source_sequence_num
+                )
+            )
+        )
+
+    @staticmethod
+    def send_message_to_database(json_data: str):
+        """
+        Send a preserialized JSON message to the database.
+
+        Note: The third key in the JSON data is assumed to be the table name
+        """
+        # Extract the table name from the JSON data
+        json_data = json.loads(json_data)
+        table_name = list(json_data.keys())[2]
+
+        logger.info(f"Adding an entry to the {table_name} table")
+        logger.trace(f"Entry: {json_data[table_name]}")
+
+        # Push the JSON data to PocketBase using the correct schema
+        DatabaseHandler.client.collection(table_name).create(json_data[table_name])
+
+# Procedures ======================================================================================
+def database_thread(thread_name: str, db_workq: mp.Queue, message_handler_workq: mp.Queue):
+    """
+    The main loop of the database handler. It subscribes to the CommandMessage collection
+    """
+
+    DatabaseHandler(thread_name, db_workq, message_handler_workq)
+
+    while 1:
+        # If there is any workq messages, process them
+        if not db_workq.empty():
+            if not process_workq_message(db_workq.get(block=True)):
+                return
```

Comment:
Make sure to also get rid of the `db_workq.empty()` check, because this is what'll make it basically spam-poll the queue as opposed to actually blocking on the .get call

Reply:
Oh right I didnt know what you meant before but I get it, is this ok with the subscribe function? will the callback still be able to let us know there is a dataabase notification??

Reply:
Subscribe should be ok, we should give it a quick test but based on a quick look at their code their event loops work with `anyio`, and we're not calling into any handlers so blocking here should be a-ok (maybe better for subscribe since that gives the subscribe function more time to run)

Reply:
idk what i just replied to.. but I tested this and it worked all good